data-prepper-plugins/opensearch/src/main/java/com/amazon/dataprepper/plugins/sink/opensearch/index/IndexManager.java

```diff
@@ -34,14 +43,98 @@
     protected RestHighLevelClient restHighLevelClient;
     protected OpenSearchSinkConfiguration openSearchSinkConfiguration;
     protected IsmPolicyManagementStrategy ismPolicyManagementStrategy;
+    protected String indexPrefix;
 
     private static final Logger LOG = LoggerFactory.getLogger(IndexManager.class);
 
+    //For matching a string that begins with a "%{" and ends with a "}".
+    //For a string like "data-prepper-%{yyyy-MM-dd}", "%{yyyy-MM-dd}" is matched.
+    private final static String TIME_PATTERN_REGULAR_EXPRESSION  = "%\\{.*?\\}";
+
+    //For matching a string enclosed by "%{" and "}".
+    //For a string like "data-prepper-%{yyyy-MM}", "yyyy-MM" is matched.
+    private final static String TIME_PATTERN_INTERNAL_EXTRACTOR_REGULAR_EXPRESSION  = "%\\{(.*?)\\}";
+
+    private Optional<DateTimeFormatter> indexTimeSuffixFormatter;
+    private final static ZoneId UTC_ZONE_ID = ZoneId.of(TimeZone.getTimeZone("UTC").getID());
+
     protected IndexManager(final RestHighLevelClient restHighLevelClient, final OpenSearchSinkConfiguration openSearchSinkConfiguration){
         checkNotNull(restHighLevelClient);
         checkNotNull(openSearchSinkConfiguration);
         this.restHighLevelClient = restHighLevelClient;
         this.openSearchSinkConfiguration = openSearchSinkConfiguration;
+        initializeIndexPrefixAndSuffix();
+    }
+
+    private void initializeIndexPrefixAndSuffix(){
+        String indexAliasFromConfig = openSearchSinkConfiguration.getIndexConfiguration().getIndexAlias();
+        indexPrefix = indexAliasFromConfig.replaceAll(TIME_PATTERN_REGULAR_EXPRESSION, "");
```

Comment:
I would recommend only setting this after validation is complete.

Reply:
+1